### DISCUSSION TITLE
Exception: getUploadUrl command is not supported.

### DISCUSSION THREAD

**Discussion by lucasMarino7** [flet-dev/flet#3383] [Q&A]
**Description**
When i runing the example code about upload files from file picker([https://github.com/flet-dev/examples/blob/main/python/controls/file-picker/file-picker-upload-progress.py](url)) and remove the web view, return this error: "Exception: getUploadUrl command is not supported.". How can i upload files from file picker in desktop app?.
Sorry for my bad english.

My code:
```python
from typing import Dict

import flet
from flet import (
    Column,
    ElevatedButton,
    FilePicker,
    FilePickerResultEvent,
    FilePickerUploadEvent,
    FilePickerUploadFile,
    Page,
    ProgressRing,
    Ref,
    Row,
    Text,
    icons,
)


def main(page: Page):
    prog_bars: Dict[str, ProgressRing] = {}
    files = Ref[Column]()
    upload_button = Ref[ElevatedButton]()

    def file_picker_result(e: FilePickerResultEvent):
        upload_button.current.disabled = True if e.files is None else False
        prog_bars.clear()
        files.current.controls.clear()
        if e.files is not None:
            for f in e.files:
                prog = ProgressRing(value=0, bgcolor="#eeeeee", width=20, height=20)
                prog_bars[f.name] = prog
                files.current.controls.append(Row([prog, Text(f.name)]))
        page.update()

    def on_upload_progress(e: FilePickerUploadEvent):
        prog_bars[e.file_name].value = e.progress
        prog_bars[e.file_name].update()

    file_picker = FilePicker(on_result=file_picker_result, on_upload=on_upload_progress)

    def upload_files(e):
        uf = []
        if file_picker.result is not None and file_picker.result.files is not None:
            for f in file_picker.result.files:
                uf.append(
                    FilePickerUploadFile(
                        f.name,
                        upload_url=page.get_upload_url(f.name, 600),
                    )
                )
            file_picker.upload(uf)

    # hide dialog in a overlay
    page.overlay.append(file_picker)

    page.add(
        ElevatedButton(
            "Select files...",
            icon=icons.FOLDER_OPEN,
            on_click=lambda _: file_picker.pick_files(allow_multiple=True),
        ),
        Column(ref=files),
        ElevatedButton(
            "Upload",
            ref=upload_button,
            icon=icons.UPLOAD,
            on_click=upload_files,
            disabled=True,
        ),
    )


flet.app(target=main, upload_dir="uploads")

```

**Issues**:
![error](https://github.com/flet-dev/flet/assets/122319328/aa12d682-50c9-4a20-ae76-cbcce3ffc207)

**Flet version**:

```
Name: flet
Version: 0.10.1
Summary: Flet for Python - easily build interactive multi-platform apps in Python
Home-page:
Author: Appveyor Systems Inc.
Author-email: hello@flet.dev
License: Apache-2.0
Location: C:\Users\conta\AppData\Local\Programs\Python\Python311\Lib\site-packages
Requires: copier, flet-runtime, packaging, qrcode, watchdog, websocket-client, websockets
Required-by:
```

**Operating system**:
Windows

**Comment by jmcrespo1960** [comment]
https://www.youtube.com/watch?v=Z0luQoMchNA
Hi! I Found this example.
Credits to Sri Edy Nurcahyo.

**Comment by umang299** [comment]
> https://www.youtube.com/watch?v=Z0luQoMchNA Hi! I Found this example. Credits to Sri Edy Nurcahyo.

This method somehow the files from ``file_picker.result`` have path attribute as None. This error is observed only when the app is run in browser view. 

``ft.app(target=main, view=ft.WEB_BROWSER)``

Kindly address this at the earliest. 

Thank you.

**Comment by ndonkoHenri** [comment]
I am just from trying your code and it works. There is just one thing I needed to set somewhere at the top of your code which is the secret key, needed to enable uploads:
```python
import os
os.environ['FLET_SECRET_KEY'] = "YOUR_SECRET_KEY"
```

**Reply by SamYuan1990** [reply]
hi @ndonkoHenri 

May I know what the key like?
I don't want to have a key... or say for development phase I don't have a key yet.
so
```
os.environ["FLET_SECRET_KEY"] = "DEFAULT_SECRET_KEY_CHANGE_IN_PRODUCTION"
... some other code
flet.app(target=main, upload_dir="uploads")
```

but it won't work either, could you please advise?

**Comment by SamYuan1990** [comment]
I tested it with macOS today... honestly, it won't work for either `FLET_SECRET_KEY` or `view=ft.WEB_BROWSER`
I got the point as there some setting for flet.app 
but I got confused, if we using default app model, which won't work for upload. if we turn to ft.WEB_BROWSER, the file picker won't work for download ? or it will work? is there any solution or alternative ?

**Reply by SamYuan1990** [reply]
I suppose this is a bug, as app start relay on no web impl, and file picker just works for web application model.
